src/framework/platform/browserplatform.cpp

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2010-2016 OTClient <https://github.com/edubart/otclient>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#ifdef __EMSCRIPTEN__
+
+#include "platform.h"
+#include <cstring>
+#include <fstream>
+#include <unistd.h>
+#include <framework/stdext/stdext.h>
+#include <framework/core/eventdispatcher.h>
+#include <errno.h>
+
+#include <sys/stat.h>
+
+void Platform::init(std::vector<std::string>& args)
+{
+    processArgs(args);
+    setDevice({ Browser, OsUnknown });
+}
+
+void Platform::processArgs(std::vector<std::string>& /*args*/)
+{
+    //nothing todo, linux args are already utf8 encoded
```

Comment:
That is not strictly true. Linux arguments does not have to be utf8, there is only 1 hard rule for linux shell arguments: they cannot contain null bytes (and they have a max length). Little test to prove it:
```php
$ cat utf8lol.php
<?php
$invalidUtf8 = "\xC2\x41";
$cmd = 'php -r "var_dump(mb_check_encoding(\$argv[1], \'UTF-8\'), \$argv[1]);" \'' . $invalidUtf8 . '\'';
system($cmd);
?>
$ php utf8lol.php
bool(false)
string(2) "�A"
```
- voila: passed a linux shell argument that was not valid UTF-8 ^^


But it is a very edge-case, no need to waste time on it here really.